BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java

```diff
@@ -46,6 +49,14 @@ public ModuleState(@JsonProperty("module") Module module,
     this.pendingBranchBuild =  filterBranchBuild(pendingBranchBuild);
   }
 
+  /* When this object is built by rosetta the missing build rows are filled with nulls,
+   * which results in objects like this being passed to ModuleState's constructor:
+   *
+   * ModuleBuild(Optional.absent(), 0, 0, 0, null, Optional.absent()...)
+   *
+   * This results in a "present" ModuleBuild, but it has no actual values, the filters here
+   * find such elements and replace them with just Optional.absent()
+   */
```

Comment:
I have confirmed this is the case and will remove the exceptions.
I'll create a new PR to add the exceptions. We'll have to re-think how we create objects for the status in order to add this checking in the constructor.